src/main/java/cr0s/warpdrive/compat/CompatEnergyControl.java

```diff
@@ -0,0 +1,246 @@
+package cr0s.warpdrive.compat;
+
+import cr0s.warpdrive.WarpDrive;
+import cr0s.warpdrive.api.IBlockTransformer;
+import cr0s.warpdrive.api.ITransformation;
+import cr0s.warpdrive.api.WarpDriveText;
+import cr0s.warpdrive.config.WarpDriveConfig;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class CompatEnergyControl implements IBlockTransformer {
+	
+	//only require rotation
+	private static Class<?> classThermalMonitor;            //Thermal Monitor
+	private static Class<?> classRemoteThermalMonitor;      //Remote Thermal Monitor
+	
+	//rotation and coordinate
+	private static Class<?> classInfoPanel;                 //(Advanced) Info Panel
+	private static Class<?> classInfoPanelExtender;         //(Advanced) Info Panel Extender
+	private static Class<?> classHoloPanel;                 //Holographic Panel
+	private static Class<?> classHoloPanelExtender;         //Holo Extender
+	
+	private static Class<?> classRangeTrigger;              //Range Trigger
+	
+	public static void register(){
```

Comment:
Consider following existing code spacing rules?
* tabulation -> space
* space before and after { } + - / etc.
* first condition on the same line as if
* then instruction always on a different line than the if
* case aligned (not indented) vs switch statement
* etc.